src/MongoDB.Driver/IAggregateFluent.cs

```diff
@@ -339,6 +339,20 @@ public interface IAggregateFluent<TResult> : IAsyncCursorSource<TResult>
         /// <returns>A Task.</returns>
         Task ToCollectionAsync(CancellationToken cancellationToken = default(CancellationToken));
 
+        /// <summary>
+        /// Appends an $unionWith stage to the pipeline.
+        /// </summary>
+        /// <typeparam name="TForeignDocument">The type of the foreign collection documents.</typeparam>
+        /// <typeparam name="TNewResult">The type of the new result.</typeparam>
+        /// <param name="foreignCollection">The foreign collection.</param>
+        /// <param name="unionWithPipeline">The unionWith pipeline.</param>
+        /// <param name="options">The options.</param>
+        /// <returns>The fluent aggregate interface.</returns>
+        IAggregateFluent<TNewResult> UnionWith<TForeignDocument, TNewResult>(
+            IMongoCollection<TForeignDocument> foreignCollection,
+            PipelineDefinition<TForeignDocument, TNewResult> unionWithPipeline = null,
+            AggregateUnionWithOptions<TForeignDocument, TNewResult> options = null);
+
```

Comment:
I think this should be:

    /// <summary>
    /// Appends an $unionWith stage to the pipeline.
    /// </summary>
    /// <typeparam name="TWith">The type of the with collection documents.</typeparam>
    /// <param name="withCollection">The with collection.</param>
    /// <param name="withPipeline">The with pipeline.</param>
    /// <returns>The fluent aggregate interface.</returns>
    IAggregateFluent<TResult> UnionWith<TWith>(
        IMongoCollection<TWith> withCollection,
        PipelineDefinition<TWith, TResult> withPipeline = null);

The input documents (from the pipeline so far) are of type TResult.

The withPipeline must convert TWith documents to TResult documents so that all the documents leaving this stage are all of the same type.

Reply:
done